app/webpacker/components/CompetitionsOverview/ListView.js

```diff
@@ -18,34 +18,37 @@ function ListView({
   hasMoreCompsToLoad,
 }) {
   const { ref: bottomRef, inView: bottomInView } = useInView();
+
   useEffect(() => {
     if (hasMoreCompsToLoad && bottomInView) {
       fetchMoreCompetitions();
     }
-  }, [bottomInView, hasMoreCompsToLoad, fetchMoreCompetitions]);
+  }, [bottomInView, hasMoreCompsToLoad, fetchMoreCompetitions, competitions]);
```

Comment:
An unused dependency is not necessarily wrong, but generally not a good sign I think.

Reply:
I get your thinking, and I'm open to suggestions for improvement. Here's the context of this hook:

Tables in the main body (i.e. the actual list of competitions) are supposed to have infinity scrolling. Usually this works by detecting whether the `ref` that we point to the end of the table is in view. The `useInView` hook from `react-intersection-observer` works as expected in this regard.

The problem is that the `useEffect` hook only triggers when one of its dependencies _changes_. If your screen is so large, that loading the next paginated batch of competitions does *not* push the footer of the table out of view (because they all fit on your current screen), the `bottomInView` variable never _changes_. It _stays_ in view but we still want to trigger the "load more" hook.

Reply:
Thanks. Makes sense. I think a brief comment in the code would be appropriate.

The only alternative that I can think of is providing some kind of `onSuccess` (or equivalent) callback to the query, which would check `bottomInView` and fetch again if necessary. I don't know if that's possible, or if it would even be better though.